### PR TITLE
Fix A11y: #44

### PR DESCRIPTION
**Pull request by blasten**
Original issue #44 
- [x] https://github.com/PolymerElements/iron-behaviors/pull/31 (comments addressed)
- [x] https://github.com/PolymerElements/iron-menu-behavior/pull/29  (comments addressed)
- [x] https://github.com/Polymer/polymer/pull/2517 by @notwaldorf

### CODE REVIEW

**Comment by samccone** [comment]
looks like we need a rebase :jack_o_lantern:

**Comment by blasten** [comment]
rebased!

**Comment by blasten** [comment]
friendly ping @cdata @notwaldorf

**Comment by cdata** [comment]
Since this fixes an issue, would it be possible to add a test that might catch regressions?

File: paper-submenu.html [hunk 1]
```diff
@@ -63,137 +63,167 @@
...
+      },
+
+      /**
+       * A handler that is called when `iron-activate` is fired.
+       *
+       * @param {CustomEvent} event An `iron-activate` event.
+       */
+      _onParentIronActivate: function(event) {
+        var parent = this.__parent;
+        if (Polymer.Gestures.findOriginalTarget(event) === parent) {
```
Comment (cdata) [inline paper-submenu.html]: Is `event.currentTarget` the same as `Polymer.Gestures.findOriginalTarget`? It might be cheaper to check that.

**Comment by cdata** [comment]
This PR is looking really nice and clean. I left you some feedback. PTAL!

**Comment by cdata** [comment]
@blasten bump

File: paper-submenu.html (hunk 1 above)
Comment (blasten) [inline paper-submenu.html]: It doesn't seem like that is the case: https://github.com/Polymer/polymer/blob/42004df48ca8324bb2b98ca2a4f68393faeb526a/src/standard/gestures.html#L195-L201  cc @azakus for confirmation.

**Comment by blasten** [comment]
@cdata I added a test PTAL

**Comment by cdata** [comment]
LGTM :+1: Thanks for bearing with me on this one!

**Comment by blasten** [comment]
🙏

**Pull request merged**